### DISCUSSION TITLE
Snap Package

### DISCUSSION THREAD

**Discussion by TCROC** [containers/podman#14360] [Ideas]
Building off of this issue: https://github.com/containers/podman/issues/14302

Would it be possible to package podman into a cross platform package manager such as Snap?  I'm not familiar at all with the internals, just brainstorming different ways to streamline it since its super cool!

(So far I'm absolutely LOVING not having to run docker as root!!! :) )

**Comment by lsm5** [comment]
IIRC, there was already a snap / flatpak created. I don't know the current status though. @TomSweeneyRedHat any idea?

**Comment by rhatdan** [comment]
This is an opensource project, if community wants to maintain SNAP or Flatpak we are fine with that.

**Comment by afbjorklund** [comment]
> (So far I'm absolutely LOVING not having to run docker as root!!! :) )

https://docs.docker.com/engine/security/rootless/

**Comment by TCROC** [comment]
> 

Ya.  I read through that, but was not very happy with that as a solution.  I definitely prefer Podman's out of the box approach.

**Comment by lsm5** [comment]
@TCROC i have restored the podman package on the kubic repo. It's still v3.4.2 though.

A Snap package would  likely have to be a community contribution as upstream podman won't be providing it.

**Comment by vrothberg** [comment]
I concur with @lsm5. This is not something upstream can realistically provide.  I am converting it into a discussion.

**Comment by TCROC** [comment]
I'm just moving this question over to here:

https://github.com/containers/podman/issues/14302#issuecomment-1224997848

I believe I mentioned it earlier and it was said the community could do it if they wanted, but I just thought I'd bring it up again:

Why is using a cross platform packaging software such as Snapcraft as an official distro not being discussed more? I believe the selling point of these packaging tools is they are supposed to solve a lot of these complexities. Is there a reason Podman wouldn't want to officially use one?

^ Keep in mind, I'm merely a naive dev who wants to use the shiny toys such as new Podman on my Ubuntu :)

@vrothberg @lsm5 

Pinging you guys on this since you seemed to have some insight.  Thanks! :)

**Reply by vrothberg** [reply]
Thanks for the ping, @TCROC!

> Why is using a cross platform packaging software such as Snapcraft as an official distro not being discussed more? 

Aren't we discussing it in this very thread?

To me the question is not about the usefulness of snapcraft or flatpaks, but more how to realistically maintain it.  If volunteers step up and do the work, it will be more than welcome.  We are not actively blocking it from happening but there is only so much/little time we have.

As mentioned in the other thread, I fully sympathize with the desire of running new releases on old distributions (e.g., Ubuntu LTS releases). But it's practically impossible to package new releases for all the distributions out there, especially older ones.

**Reply by lsm5** [reply]
Speaking from the Red Hat/Fedora POV (kinda sorta different from Podman the project), Fedora is top priority so there's no way we're giving up on Fedora packaging for the foreseeable future. Now, if I can generate debian packages using some additions to Fedora's own packaging sources, that means a lot less additional maintenance work for me, which is what led me to give up on the prior `Kubic Stable` repo in favour of the `Kubic Unstable` repo.

Now, snap packaging may be distro-agnostic but it would mean additional maintenance and snap package also doesn't mean anything officially to Fedora (and CentOS and RHEL ). So, I'd rather not own snap packaging myself, and other Podman members have their hands full with other things already.

That's why we're leaving snap to the community. In fact, there was someone a few years ago who did create snaps for skopeo and buildah I think, don't know about podman. And also don't know current status of those snaps.

HTH.

**Reply by afbjorklund** [reply]
Even on Ubuntu, which does use `snap`, it makes more sense for something like Spotify than for a low-level tool like Podman ?

In fact, there's a lot of problems with the sandboxing of the Docker snap - have to redirect users to install "real" packages instead.
They can choose between using the packages from the distribution or to use the packages from the vendor, that matters less...

But all the limitations of snap/snapd and `/snap` makes it easier to just use a regular package. Unless you are Canonical, perhaps.
